### DISCUSSION TITLE
Doesn't actually create branches?

### DISCUSSION THREAD

**Discussion by ScottChapman** [repository-settings/app#460] [Q&A]
I'm trying to set this up for (on site deployment for our GHE) a test org. And one of the things I wanted to be able to do was to use `main` as the default branch rather than `master`.
So, I created a section in my YAML for the `main` branch and included some branch protection rules. But I get an error saying that the branch `main` was not found. Should that have been created?

Also, I did not see a mechanism to specify what the default branch was?

**Comment by travi** [comment]
could you describe your use case a bit? setting the default branch name is supported, but i'm curious about your expectation to create the branch. 

settings accomplishes the current approach in the api call to update the settings of the repo: https://docs.github.com/en/rest/reference/repos#update-a-repository. the change supported here is simply switching the default from one existing branch to a different existing branch, like https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/changing-the-default-branch

alternatively, it is possible to rename an existing branch to a different name without switching to a separate, already existing branch, like https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/renaming-a-branch. additional guidance can be found in https://github.com/github/renaming. if you have lots of repos where you want to rename the default branch, you might consider using octoherd: https://github.com/octoherd/script-rename-master-branch-to-main

however, i wouldnt expect a tool like settings to create a branch in either of these scenarios. could you expand on why you would expect a branch to be created and what part settings would play in such a scenario?

<blockquote><div><strong><a href="https://docs.github.com/en/rest/reference/repos#update-a-repository">Repositories - GitHub Docs</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/changing-the-default-branch">Changing the default branch - GitHub Docs</a></strong></div><div>If you have more than one branch in your repository, you can configure any branch as the default branch.</div></blockquote>
<blockquote><div><strong><a href="https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/renaming-a-branch">Renaming a branch - GitHub Docs</a></strong></div><div>You can change the name of a branch in a repository.</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/31a0e592b40963a46907dd0eb784d799dc305d3ed7a0ab62f1e7492df749b189/github/renaming" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/github/renaming">github/renaming</a></strong></div><div>Guidance for changing the default branch name for GitHub repositories - github/renaming</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/c31669aa1528db82f5f338b5cf4b5d51fca131d179f5512ca6d91e7d8f68a8a5/octoherd/script-rename-master-branch-to-main" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/octoherd/script-rename-master-branch-to-main">octoherd/script-rename-master-branch-to-main</a></strong></div><div>Renames the default branch to main if it is currently set to master - octoherd/script-rename-master-branch-to-main</div></blockquote>

**Reply by sebandraos** [reply]
Hi @travi , 
Whilst I entirely agree with you that creating branches is beyond the scope of an (excellent) app like this, what is the expected behaviour if the branch named for the protection settings doesn't exist? We build most of our repos from templates and use a couple of different branch naming conventions depending on the age of the repo and team using it.
Ideally we'd like to list all the possible (3-4) main branch names in the protection rules but as only one of those is going to exist when the repo is created, are the rules going to be applied to that branch or will it fail because one or more of the branches aren't found?
Thanks
